source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,52 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+
+  if ((connection.ssl() && connection.ssl()->isMtls()) ||
+      config.forwardClientCert() == Http::ForwardClientCertType::AlwaysForwardOnly) {
+    std::string clientCertDetails;
+    if (config.forwardClientCert() == Http::ForwardClientCertType::AppendForward ||
+        config.forwardClientCert() == Http::ForwardClientCertType::SanitizeSet) {
+      // In these cases, the client cert in the current hop should be set into the XFCC header.
+      if (!connection.ssl()->uriSanLocalCertificate().empty()) {
+        clientCertDetails += "BY=" + connection.ssl()->uriSanLocalCertificate();
+      }
+      if (!connection.ssl()->sha256PeerCertificateDigest().empty()) {
+        clientCertDetails += ";Hash=" + connection.ssl()->sha256PeerCertificateDigest();
+      }
+      for (auto detail : config.setClientCertDetails()) {
+        if (detail == Http::ClientCertDetailsType::Subject) {
+          // TODO. Get Cert Subject.
+        } else if (detail == Http::ClientCertDetailsType::SAN) {
+          // Currently, we only support a single SAN field.
+          clientCertDetails += ";SAN=" + connection.ssl()->uriSanPeerCertificate();
```

Comment:
We need to document this header format and the configuration of this features in https://github.com/lyft/envoy/tree/master/docs somewhere.